### ISSUE TITLE
Media tab not working for certain Twitter accounts (possibly accounts that post sensitive media)

### ISSUE THREAD

**Issue by the01player** [TwidereProject/Twidere-Android#1335]
## Expected behavior
When clicking on the media tab for any twitter account, you would expect to see their recent media. If you cannot, there should be a toggle-able setting to fix that.

## Actual behavior
Media is not showing up at all for some Twitter accounts, even after attempting to refresh the tab. (examples https://drive.google.com/drive/folders/1-2V-KUoZ6q7Yx03wrL4S6lyHxcQxTe2k?usp=sharing )
I suspect that sensitive posts (such as 18+ or offensive content) are not loading in the media tab, however I have not checked enough accounts and posts to verify that these are the only type of affected posts. Additionally, even though these posts will not show up in the Media tab, they are able to be seen everywhere else, such as your homepage, the Tweets tab, or the Likes tab because I have enabled the "Display sensitive content" setting.

## Steps to reproduce
01. Find an account that posts sensitive content.
02. See if that content is viewable in the media tab.
03. Compare that to a twitter account that doesn't post any sensitive content and see if their media tab loads their posts properly.

## Extra info

Android version: 10

Network type:

- [X] Cellular
- [X] Wi-Fi

App version: 4.1.4

Build variant:

- [ ] Google
- [X] F-Droid

Micro-blogging service:

- [X] Twitter
- [ ] Fanfou
- [ ] GNU Social / StatusNet
- [ ] Mastodon

App settings:

- [ ] Custom API
- [ ] Custom consumer key/secrets
- [ ] [Official keys](https://gist.github.com/mariotaku/5465786)
- [ ] Using [Thumbor](https://github.com/thumbor/thumbor)
- [ ] Connected through proxy
- [X] Media preview enabled

**Comment by Wyvernix** [comment]
I am also seeing this behavior. It appears to be related to either accounts marked "sensitive" or posts marked "sensitive" (I have not tested the difference). I also have the always display sensitive content setting marked.

Additionally, I will see a few image posts in the media tab, but they appear to be only 2-3 posts from the most recent tweets. There is no way to load more entries on the media tab. Also like the OP, the images in the media tab do not appear to be sensitive tweets.

I have been using this app for a long time, and first noticed this occurring when switching from the (un)official Twitter keys to the normal API keys. When using the Twitter keys this issue did not occur. This may be a limitation of Twitter API?

**Comment by deinodon** [comment]
same, seems kind of random though, it does say "recent media" underneath as well, so it wont even load past a certain date

**Comment by Nachtalb** [comment]
This still seems to be a problem. @Tlaster I know you are working hard on twidere x, but atm twidere x can't download videos (or at least I don't know how) but the media tab works there. So could you either fix this crucial thing here or try to implement the video download in twidere x, please

**Comment by the01player** [comment]
@Nachtalb I don't think this app is being actively developed anymore, most effort is being put into Twidere-X from what I can see. 
I'm sure they would rather fix the downloading videos problem of Twidere-X then fix this bug (sounds like it would be an easy fix too compared to this).

**Comment by Nachtalb** [comment]
@the01player 
> Sorry for too late since I'm focusing on Twidere X quite some time and due to the lack of man (just me ATM) the old Twidere can not get update as frequently as the new Twidere X.

from the last twindere release 10 days ago. So it is still being developed. 

But either way, I think I found the problem. It's called twitter being a little ...https://developer.twitter.com/en/docs/twitter-api/search-overview  If you don't pay them hundreds, even thousands of $$ 💵  they will only give you 7 days of search. Every time we open a users Twitter page we do a search request for the media tab https://github.com/TwidereProject/Twidere-Android/blob/4065eddb321c2bab48d628e943fb20600b4d494d/twidere/src/main/kotlin/org/mariotaku/twidere/loader/statuses/MediaTimelineLoader.kt#L109

In other words... this ain't gonna get fixed, as there is no bug. (though I guess you could do some trickery with loading the timeline and extracting the media tweets from there https://developer.twitter.com/en/docs/twitter-api/v1/tweets/timelines/overview)

![image](https://user-images.githubusercontent.com/9467802/110716950-750bf800-8208-11eb-952e-cfe51d894521.png)
![image](https://user-images.githubusercontent.com/9467802/110717113-bc928400-8208-11eb-835f-85681b78dd08.png)
![image](https://user-images.githubusercontent.com/9467802/110717136-c6b48280-8208-11eb-8151-4d3b2dfacfd3.png)

**Comment by the01player** [comment]
@Nachtalb Ah that sucks... Understandable since that does cost twitter resources, but that is very expensive.

Also, I would have to do more testing to see if it really is the 7 days issue, or if its an issue with sensitive accounts, but I don't use social media on my phone anymore so this doesn't matter to me much, but I hope a solution can be found nonetheless.

**Comment by Nachtalb** [comment]
Idk this setting may also have an effect on the search through the API. But from my testing it's the 7 days thingy.

![Screenshot_20210311-015525_Chrome](https://user-images.githubusercontent.com/9467802/110719276-dd5cd880-820c-11eb-9a69-e8ab028f3e01.png)

**Comment by Tlaster** [comment]
>@Nachtalb I don't think this app is being actively developed anymore, most effort is being put into Twidere-X from what I can see.
I'm sure they would rather fix the downloading videos problem of Twidere-X then fix this bug (sounds like it would be an easy fix too compared to this).



Twidere is still being developed until the new Twidere X can replace the old Twidere completely, and we're hiring so that we can have more people to maintain the old Twidere (not just me).
As for the issue, it is the 7 days limit from Twitter, so instead of using search API, Twidere X just fetching the user timeline and extracting the media from it, the old Twidere can be upgraded to use the same strategy, but will be later.
